src/object/obj_tx.c

```diff
@@ -392,6 +392,15 @@ dc_tx_alloc(daos_handle_t coh, daos_epoch_t epoch, uint64_t flags,
 	return rc;
 }
 
+static inline void
+dc_tx_bulk_free(crt_bulk_t *hdl)
+{
+	if (hdl != NULL && *hdl != CRT_BULK_NULL) {
```

Comment:
Looks 'hdl' can't be NULL, we'd assert(hdl != NULL)?

Reply:
For the case of dc_tx_bulk_free() maybe called repeatedly, especially when error handing for some failure before sending out CPD RPC.